service_container/injection_types.rst

```diff
@@ -100,6 +100,112 @@ working with optional dependencies. It is also more difficult to use in
 combination with class hierarchies: if a class uses constructor injection
 then extending it and overriding the constructor becomes problematic.
 
+Immutable-setter Injection
+--------------------------
+
+Another possible injection is to use a method which returns a separate instance
+by cloning the original service, this approach allows you to make a service immutable::
+
+    // ...
```

Comment:
Please replace this line by a a fake use statement for Mailerinterface

Reply:
Added :)